src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/NameFormatter.cs

```diff
@@ -98,37 +112,45 @@ private static void BuildClassName(StringBuilder builder, NameCache cache, ulong
             }
         }
 
-        public static void BuildGenericParameters(StringBuilder builder, NameCache cache, ulong[] parameters)
+        public static void BuildGenericTypeNames(StringBuilder builder, NameCache cache, ulong[] parameters, TypeFormat typeFormat = TypeFormat.FullName)
+        {
+            IList<string> typeNames = GetTypeNames(cache, parameters, typeFormat);
+
+            WriteTypeNamesList(builder, typeNames);
+        }
+
+        public static void WriteTypeNamesList(StringBuilder builder, IList<string> typeNames, char startChar = GenericStart, char endChar = GenericEnd, char separationChar = GenericSeparator)
```

Comment:
Rather than just default values, could you add two more methods that are explicit about what they are formatting: `BuildGenericArgTypes` and `BuildMethodParameterTypes` that will specify these three pieces?